autoload/ale/lsp_linter.vim

```diff
@@ -271,6 +271,29 @@ function! ale#lsp_linter#OnInit(linter, details, Callback) abort
         call ale#lsp#NotifyForChanges(l:conn_id, l:buffer)
     endif
 
+    " Tell the relevant buffer that the LSP has started via an autocmd.
+    if l:buffer > 0
+        if l:buffer == bufnr('')
+            silent doautocmd <nomodeline> User ALELSPStarted
+        else
+            execute 'augroup ALELSPStartedGroup' . l:buffer
+                autocmd!
+
+                execute printf(
+                \   'autocmd BufEnter <buffer=%d>'
+                \       . ' doautocmd <nomodeline> User ALELSPStarted',
+                \   l:buffer
+                \)
+
+                " Replicate ++once behavior for backwards compatibility.
+                execute printf(
+                \   'autocmd! ALELSPStartedGroup%d BufEnter <buffer>',
```

Comment:
This should be `autocmd! ALELSPStartedGroup BufEnter <buffer=%d>`, and see other comment.

Reply:
I wasn't able to get that exact suggestion to work. I would run into errors like this:

```
    (12/12) [EXECUTE] (X) Vim(autocmd):E216: No such event: <buffer>
      > BufEnter Autocommands for "<buffer=3>"
```

I fixed the syntax and retained the buffer-specific autocommand group, and that does seem to act as intended.

Reply:
You know, I might have missed that the group was created with the buffer number attached, in which case your approach is correct. My bad.